bin/dfx-software-nns-dapp-install

```diff
@@ -0,0 +1,82 @@
+
+ND_WASM_FILENAME="nns-dapp.wasm"
+ND_WASM_URL="$(dfx-software-nns-dapp-ci-wasm-url --release "${DFX_ND_RELEASE}")"
+curl --retry 5 --fail -sSL "${ND_WASM_URL}" >"$ND_WASM_FILENAME"
+
+: There is no standardization for how args are constructed. We can use a pre-built set of arguments as long as all canister IDs, feature flags etc are exactly as in the upstream build. E.g. there is no standardized way of passing in the IDs of a list of canisters.
+case "$DFX_NETWORK" in
+ic | mainnet) ND_ARG_FILENAME=nns-dapp-arg-mainnet.bin ;;
+*) ND_ARG_FILENAME=nns-dapp-arg-local.bin ;;
+esac
+
+if false; then # The PR for publishing arguments hasn't been merged yet
+  ND_ARG_URL="${ND_WASM_URL%/*}/${ND_ARG_FILENAME}"
+  curl --retry 5 --fail -sSL "${ND_ARG_URL}" >"$ND_ARG_FILENAME"
+  cp "$ND_ARG_FILENAME" nns-dapp-arg.wasm
+else
+  echo "WARNING:  Using a toy config.  Please get the real arguments published and use those." >&2
+  cat <<-EOF >"$ND_ARG_FILENAME"
+	(opt record{
+	  args = vec {
+	    record{ 0="CYCLES_MINTING_CANISTER_ID"; 1="rkp4c-7iaaa-aaaaa-aaaca-cai" };
+	    record{ 0="DFX_NETWORK"; 1="local" };
+	    record{ 0="FEATURE_FLAGS"; 1="{\"ENABLE_CKBTC\":false,\"ENABLE_CKTESTBTC\":false,\"ENABLE_SIMULATE_MERGE_NEURONS\":false,\"ENABLE_SNS_AGGREGATOR\":true,\"ENABLE_SNS_VOTING\":true}" };
+	    record{ 0="FETCH_ROOT_KEY"; 1="true" };
+	    record{ 0="GOVERNANCE_CANISTER_ID"; 1="rrkah-fqaaa-aaaaa-aaaaq-cai" };
+	    record{ 0="HOST"; 1="http://localhost:8080" };
+	    record{ 0="IDENTITY_SERVICE_URL"; 1="http://qhbym-qaaaa-aaaaa-aaafq-cai.localhost:8080" };
+	    record{ 0="LEDGER_CANISTER_ID"; 1="ryjl3-tyaaa-aaaaa-aaaba-cai" };
+	    record{ 0="OWN_CANISTER_ID"; 1="qsgjb-riaaa-aaaaa-aaaga-cai" };
+	    record{ 0="OWN_CANISTER_URL"; 1="http://qsgjb-riaaa-aaaaa-aaaga-cai.localhost:8080" };
+	    record{ 0="ROBOTS"; 1="" };
+	    record{ 0="SNS_AGGREGATOR_URL"; 1="http://rwlgt-iiaaa-aaaaa-aaaaa-cai.localhost:8080" };
+	    record{ 0="TVL_CANISTER_ID"; 1="" };
+	    record{ 0="WASM_CANISTER_ID"; 1="qaa6y-5yaaa-aaaaa-aaafa-cai" };
+	  };
+	})
+EOF
+fi
+
+: "If the canister is not in dfx.json, it needs to be added."
+# TODO: Make a command that imports just the nns-dapp.
+dfx-nns-import
+
+: "If the canister was remote, it isn't any more."
+# Note: In previous versions of dfx, a __default remote entry did not prevent a canister from being created.  Now it does.
+for network in "${DFX_NETWORK}" __default; do
+  DFX_NETWORK="$network" jq 'del(.canisters["nns-dapp"].remote.id[env.DFX_NETWORK])' dfx.json | sponge dfx.json
+done
+
+: "Create canister, if needed"
+: "... it is easier to ask forgiveness than permission"
+dfx canister create nns-dapp --network "$DFX_NETWORK"
+: "Check that the canister seems to be present and healthy"
+if dfx canister info nns-dapp --network "$DFX_NETWORK"; then
+  echo "Canister seems to exist."
+else
+  echo "Failed to create canister.  Please make sure that any old, dead canisters have been removed." >&2
```

Comment:
Is there a command to remove an old canister?
If yes, what is it?
If no, why would we tell people to do a thing that can't be done?

Reply:
There is a command to delete a canister, but the canister continues to exist with a tombstone on it.  But I think you mean, how to tell dfx that a canister in dfx.json has not been deployed to a network?  I know of no command to do this and no longer know a of a reliable way of doing this.  As far as I understand it, there are three places where dfx may store a canister ID locally.  In dfx.json as a remote for the local network or as a default remote, in .dfx/ in the local repository, and I believe that there is a new location in the root, but I haven't bumped into that yet.  I am not an authority on dfx.  I used to modify these files directly, but with recent dfx that causes instability.  So now I really have no idea what the "right" way is, apart from asking the sdk team and reading manuals.  This PR is not here to provide such a command.

Reply:
> If no, why would we tell people to do a thing that can't be done?

Reply:
Maybe this would work as a message:
"dfx was prevented from creating a the nns-dapp canister.  This can be due to a variety of reasons.  Please read any error messages above, refer to the manual (https://internetcomputer.org/docs/current/home) and ask on the forum (https://forum.dfinity.org/) if you need help."